controllers/provider-openstack/pkg/apis/openstack/validation/validation_providerprofile.go

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2018 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validation
+
+import (
+	"fmt"
+	"time"
+
+	apisopenstack "github.com/gardener/gardener-extensions/controllers/provider-openstack/pkg/apis/openstack"
+
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// ValidateProviderProfileConfig validates a ProviderProfileConfig object.
+func ValidateProviderProfileConfig(providerProfile *apisopenstack.ProviderProfileConfig) field.ErrorList {
```

Comment:
Should we also have a validation for `DNSServers`, e.g. `net.ParseIP(dnsAddr) != nil` ?

Reply:
Done, PTAL again.